prov/efa/src/efa_user_info.c

```diff
@@ -325,23 +325,9 @@ int efa_user_info_alter_rxr(struct fi_info *info, const struct fi_info *hints)
 		 * which means FI_MR_HMEM implies FI_MR_LOCAL for device buffer.
 		 */
 		if (hints->caps & FI_HMEM) {
-			if (ofi_hmem_p2p_disabled()) {
+			if (!efa_device_support_rdma_read() && cuda_get_xfer_setting() != CUDA_XFER_ENABLED) {
 				EFA_WARN(FI_LOG_CORE,
-					"FI_HMEM capability currently requires peer to peer support, which is disabled.\n");
-				return -FI_ENODATA;
-			}
-			//TODO: remove the rdma checks once FI_HMEM w/o p2p is supported
-
-			if (!efa_device_support_rdma_read()) {
-				EFA_WARN(FI_LOG_CORE,
-				        "FI_HMEM capability requires RDMA, which this device does not support.\n");
-				return -FI_ENODATA;
-
-			}
-
-			if (!rxr_env.use_device_rdma) {
-				EFA_WARN(FI_LOG_CORE,
-				        "FI_HMEM capability requires RDMA, which is turned off. You can turn it on by set environment variable FI_EFA_USE_DEVICE_RDMA to 1.\n");
+					"FI_HMEM capability requires RDMA by default, which this device does not support.\n");
```

Comment:
How do we feel about this warning message? Would it be a good idea to add something about enable `FI_HMEM_CUDA_ENABLE_XFER` if the user wishes to use non-p2p?

Reply:
Yes. I think we need update  the warning message to add FI_HMEM_CUDA_ENABLE_XFER

Reply:
You dropped rxr_env.use_device_rdma.

Reply:
ack, see latest push